libs/wingsdk/src/cloud/test-runner.ts

```diff
@@ -0,0 +1,174 @@
+import { Construct, IConstruct } from "constructs";
+import { Function } from "./function";
+import { fqnForType } from "../constants";
+import { App } from "../core/app";
+import { Resource } from "../core/resource";
+
+/**
+ * Global identifier for `TestRunner`.
+ */
+export const TEST_RUNNER_FQN = fqnForType("cloud.TestRunner");
+
+/**
+ * Properties for `TestRunner`.
+ */
+export interface TestRunnerProps {}
+
+/**
+ * Represents a test engine.
+ *
+ * @inflight `@winglang/sdk.cloud.ITestRunnerClient`
+ */
+export abstract class TestRunner extends Resource {
+  /**
+   * Create a new test engine.
+   * @internal
+   */
+  public static _newTestRunner(
+    scope: Construct,
+    id: string,
+    props: TestRunnerProps = {}
+  ): TestRunner {
+    return App.of(scope).newAbstract(TEST_RUNNER_FQN, scope, id, props);
+  }
+
+  /**
+   * Returns whether a construct represents a runnable test.
+   * @param c A construct.
+   * @returns Whether the construct is a test.
+   */
+  public static isTest(c: IConstruct): boolean {
```

Comment:
Maybe in a subsequent PR, but I'd like us to change how tests are modeled in apps.

Instead of cloud.Functions with special IDs, let's create a cloud.Test resource and only materialize it into a Function in the environment in which it will be executed.

This means that only a single test cloud function will be defined in each environment.

Reply:
I like the idea

Reply:
#1972